maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/JarManifestForkConfiguration.java

```diff
@@ -111,7 +114,17 @@ private File createJar( @Nonnull List<String> classPath, @Nonnull String startCl
             for ( Iterator<String> it = classPath.iterator(); it.hasNext(); )
             {
                 File file1 = new File( it.next() );
-                String uri = file1.toURI().toASCIIString();
+                String uri;
+                try
+                {
+                    uri = URI.create( parent.relativize( file1.toPath() ).toString() ).toASCIIString();
```

Comment:
There is a reason I suggested the three-arg `URI` constructor rather than `create`. You want to _construct_ a URI from a raw path, not _parse_ a URI.

```
jshell> URI.create("../../a b/c.jar")
|  java.lang.IllegalArgumentException thrown: Illegal character in path at index 7: ../../a b/c.jar
|        at URI.create (URI.java:883)
|        at (#6:1)

jshell> new URI(null, "../../a b/c.jar", null)
$7 ==> ../../a%20b/c.jar
```

Reply:
Ack, my bad. Thanks.